src/sys/signal.rs

```diff
@@ -6,44 +6,125 @@ use {Errno, Result};
 use std::mem;
 use std::ptr;
 
-pub use libc::{
-    SIGHUP,
-    SIGINT,
-    SIGQUIT,
-    SIGILL,
-    SIGABRT,
-    SIGFPE,
-    SIGKILL,
-    SIGSEGV,
-    SIGPIPE,
-    SIGALRM,
-    SIGTERM,
-    SIGTRAP,
-    SIGIOT,
-    SIGBUS,
-    SIGSYS,
-    SIGURG,
-    SIGSTOP,
-    SIGTSTP,
-    SIGCONT,
-    SIGCHLD,
-    SIGTTIN,
-    SIGTTOU,
-    SIGIO,
-    SIGXCPU,
-    SIGXFSZ,
-    SIGVTALRM,
-    SIGPROF,
-    SIGWINCH,
-    SIGUSR1,
-    SIGUSR2,
-};
-
-// This doesn't always exist, but when it does, it's 7
-pub const SIGEMT: libc::c_int = 7;
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum Signal {
+    SIGHUP = libc::SIGHUP as isize,
+    SIGINT = libc::SIGINT as isize,
+    SIGQUIT = libc::SIGQUIT as isize,
+    SIGILL = libc::SIGILL as isize,
+    SIGTRAP = libc::SIGTRAP as isize,
+    SIGABRT = libc::SIGABRT as isize,
+    SIGBUS = libc::SIGBUS as isize,
+    SIGFPE = libc::SIGFPE as isize,
+    SIGKILL = libc::SIGKILL as isize,
+    SIGUSR1 = libc::SIGUSR1 as isize,
+    SIGSEGV = libc::SIGSEGV as isize,
+    SIGUSR2 = libc::SIGUSR2 as isize,
+    SIGPIPE = libc::SIGPIPE as isize,
+    SIGALRM = libc::SIGALRM as isize,
+    SIGTERM = libc::SIGTERM as isize,
+    #[cfg(not(target_os = "macos"))]
+    SIGSTKFLT = libc::SIGSTKFLT as isize,
+    SIGCHLD = libc::SIGCHLD as isize,
+    SIGCONT = libc::SIGCONT as isize,
+    SIGSTOP = libc::SIGSTOP as isize,
+    SIGTSTP = libc::SIGTSTP as isize,
+    SIGTTIN = libc::SIGTTIN as isize,
+    SIGTTOU = libc::SIGTTOU as isize,
+    SIGURG = libc::SIGURG as isize,
+    SIGXCPU = libc::SIGXCPU as isize,
+    SIGXFSZ = libc::SIGXFSZ as isize,
+    SIGVTALRM = libc::SIGVTALRM as isize,
+    SIGPROF = libc::SIGPROF as isize,
+    SIGWINCH = libc::SIGWINCH as isize,
+    SIGIO = libc::SIGIO as isize,
+    #[cfg(not(target_os = "macos"))]
+    SIGPWR = libc::SIGPWR as isize,
+    SIGSYS = libc::SIGSYS as isize,
+    #[cfg(target_os = "macos")]
+    SIGEMT = libc::SIGEMT as isize,
+    #[cfg(target_os = "macos")]
+    SIGINFO = libc::SIGEMT as isize,
+}
+
```

Comment:
The following type and function are only needed internally. They are used in the implementation of the iterator and in `wait.rs`. Because of the latter I hade to make them public (for now).

I propose putting the whole code into a private module (src/internal/sys/signal.rs) and then reexport what is supposed to be public here.